lib/proc_control.h

```diff
@@ -23,6 +25,54 @@
 #include <windows.h>
 #endif
 
+#ifdef _WIN32
+#include "boinc_win.h"
+extern int boinc_thread_cpu_time(HANDLE thread_handle, double& cpu);
+extern int boinc_process_cpu_time(HANDLE process_handle, double& cpu);
+#else
+// setpriority(2) arg to run in background
+//
+static const int PROCESS_IDLE_PRIORITY = 19;
+static const int PROCESS_MEDIUM_PRIORITY = 10;
+static const int PROCESS_NORMAL_PRIORITY = 0;
+static const int PROCESS_ABOVE_NORMAL_PRIORITY = -10;
+static const int PROCESS_HIGH_PRIORITY = -15;
+static const int PROCESS_REALTIME_PRIORITY = -20;
+extern double linux_cpu_time(int pid);
+#endif
+
+// define type for references to a process.
+// Win also has an integer PID; that's not what we mean here
+//
+#ifdef _WIN32
+#define PROCESS_ID  HANDLE
+#else
+#define PROCESS_ID  int
+#endif
+
+extern int run_program(
+    const char* dir,        // directory to run program in; NULL if current dir
+    const char* file,       // path of executable
+    int argc,
+    char *const argv[],     // cmdline args, UNIX-style
+    PROCESS_ID&             // ID of child process
+);
+extern int kill_process(PROCESS_ID);
+extern int get_exit_status(PROCESS_ID, int& status, double dt);
```

Comment:
Could make this `dt=-1.0`, to avoid the caller needing explicitly to specify the timeout when they want an infinite wait?

Reply:
Turns out that's never used.